### PR TITLE
SubTaskList and its child-components: Unit Tests, Refactoring

### PR DESCRIPTION
**Pull request by VitFL**
Unit Tests:
- Covered SubTaskList component with Unit Tests;
- Covered SubTaskItem child-component with Unit Tests;
- Covered SubTaskProgressBar child-component with Unit Tests;
- Covered SubTaskAdd child-component with Unit Tests;

Refactoring: 
- child-components moved to separate folders inside SubTaskList
- moved progressBarVariant function from SubTaskProgressBar component into separate file.
- if SubTaskProgressBar doesn't receive subtaskList in props, subtaskList now will be an empty array, instead of undefined.
- moved getSubtaskListAsArray function from SubTaskList component to separate file. 
- added check for empty taskRef in componentDidMount() lifecycle method of SubTaskList component.
- changed class name of SubTaskItem component to a proper one (same as the file name);
- changed class name of SubTaskProgressBar component to a proper one (same as the file name);
- some other minor refactoring...

### CODE REVIEW

File: src/components/SubTaskList/SubTaskList.jsx [hunk 1]
```diff
@@ -17,12 +18,13 @@ export default class SubTaskList extends Component {
```
Comment (mishalisovskiy) [inline src/components/SubTaskList/SubTaskList.jsx]: This is a rather cumbersome way to assign props to a state. Could you consider the following pattern?
`constructor(props) {
    super(props);

    this.state = {
      taskRef = props.taskRef;
      subtaskList: [];
    }
}`

Then, using destructuring, you can easily access taskRef without redundant binding to a contextual `this`.

File: src/components/SubTaskList/SubTaskProgressBar/SubTaskProgressBar.jsx [hunk 2]
```diff
@@ -1,36 +1,25 @@
 import React, { Component } from 'react';
 import { Row, Col, ProgressBar } from 'react-bootstrap';
+import getProgressBarVariant from './getProgressBarVariant';
 
-class SubTaskAdd extends Component {
+class SubTaskProgressBar extends Component {
   render() {
-    const { subtaskList } = this.props;
+    const subtaskList = this.props.subtaskList || [];
```
Comment (mishalisovskiy) [inline src/components/SubTaskList/SubTaskProgressBar/SubTaskProgressBar.jsx]: You can still use destructuring and assign a default value to a variable. Here is a way to do this.
`const { subtaskList = [] } = this.props;`

File: src/components/SubTaskList/SubTaskList.jsx (hunk 1 above)
Comment (VitFL) [reply src/components/SubTaskList/SubTaskList.jsx]: Yea, I like the pattern you proposed, except typos like ";" and "=" inside object ;)

Comment (mishalisovskiy) [reply src/components/SubTaskList/SubTaskList.jsx]: True xD. 1 am isn't the best time to code

File: src/components/SubTaskList/SubTaskProgressBar/SubTaskProgressBar.jsx (hunk 2 above)
Comment (VitFL) [reply src/components/SubTaskList/SubTaskProgressBar/SubTaskProgressBar.jsx]: Thanks for the advice! Your variant is definitely more readable and shorter =)

**Comment by mishalisovskiy** [comment]
👍🏻

**Comment by VitFL** [comment]
@mishalisovskiy , Thank you for your review and suggestions 👍

**Pull request merged**